source.go

```diff
@@ -83,20 +107,38 @@ func (s *MultipleSource) Close() error {
 
 func (s *MultipleSource) createSubscribeHandler(prefixSize int, dist *distribute) nats.MsgHandler {
 	return func(msg *nats.Msg) {
+		// call the callback for on subscription received. this will allow users to plugin metrics if they desire.
+		s.opts.onSubscriptionReceived(msg)
+
 		if 0 < prefixSize {
 			if ok := dist.Publish(msg.Subject[0:prefixSize], msg); ok != true {
 				s.logger.Printf("warn: failed to publish: %s", msg.Subject)
+				return
+			}
+
+			if s.opts.onMessageRelayed != nil {
+				s.opts.onMessageRelayed(msg)
 			}
+
 			return
 		}
 
 		if ok := dist.Publish(msg.Subject, msg); ok != true {
 			s.logger.Printf("warn: failed to publish: %s", msg.Subject)
+			return
```

Comment:
Would it be a good idea to have a third extensibility point here for `onMessageRelayFailed` to track failed attempts to relay?

Reply:
Yeah it could be useful, I have added it.